src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java

```diff
@@ -237,19 +251,29 @@ private void validatePKChunk(FailureCollector collector) {
 
     if (!containsMacro(SalesforceSourceConstants.PROPERTY_QUERY) && !Strings.isNullOrEmpty(query)) {
       if (SalesforceQueryParser.isRestrictedPKQuery(query)) {
-          collector.addFailure(
-            String.format("SOQL Query contains restricted clauses when PK Chunk is Enabled. Unsupported query: '%s'.",
-                          query),
-            "Set Enable PK Chunk to false, because 'WHERE' is the only supported conditions clause.")
-            .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
-        }
+        collector.addFailure(
+          String.format("SOQL Query contains restricted clauses when PK Chunk is Enabled. Unsupported query: '%s'.",
+                        query),
+          "Set Enable PK Chunk to false, because 'WHERE' is the only supported conditions clause.")
+          .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
+      }
 
-      String sObject = SalesforceQueryParser.getObjectDescriptorFromQuery(query).getName();
-      checkForPKSupportedObject(sObject, collector);
+      // If a parent object is defined then use that to check for PK support, otherwise use the object from the query
+      if (!containsMacro(SalesforceSourceConstants.PROPERTY_PARENT_NAME) && !Strings.isNullOrEmpty(getParent())) {
+        checkForPKSupportedObject(getParent(), collector);
+      } else {
+        String sObject = SalesforceQueryParser.getObjectDescriptorFromQuery(query).getName();
+        checkForPKSupportedObject(sObject, collector);
+      }
     }
 
+    // If a parent object is defined then use that to check for PK support, otherwise use the object from the config
     if (!containsMacro(SalesforceSourceConstants.PROPERTY_SOBJECT_NAME) && !Strings.isNullOrEmpty(getSObjectName())) {
-      checkForPKSupportedObject(getSObjectName(), collector);
+      if (!containsMacro(SalesforceSourceConstants.PROPERTY_PARENT_NAME) && !Strings.isNullOrEmpty(getParent())) {
+        checkForPKSupportedObject(getParent(), collector);
+      } else {
+        checkForPKSupportedObject(getSObjectName(), collector);
```

Comment:
same comment here. we're using the object from config if parent name is a macro.

Reply:
same response as above